azurerm/_modules/aks/node_pool/main.tf

```diff
@@ -0,0 +1,28 @@
+data "azurerm_kubernetes_cluster" "current" {
+  name                = var.cluster_name
+  resource_group_name = var.resource_group
+}
+
+resource "azurerm_kubernetes_cluster_node_pool" "current" {
+  name                  = var.node_pool_name
+  kubernetes_cluster_id = data.azurerm_kubernetes_cluster.current.id
+  enable_auto_scaling   = var.enable_auto_scaling
+  max_count             = var.enable_auto_scaling ? var.max_count : null
```

Comment:
Currently, you set defaults in different ways in multiple locations. Over time, I came to the conclusion, that it's best to only implement the defaults in the configuration.tf locals. And set no defaults for the inner module's variables. This ensures, that the outer module calls the inner module with all available variables.

In your case, the third layer of logic to set defaults is the conditionals here around the variables. It may be worth giving this a brief though, if it can be refactored to all be done in configuration.tf just once.

I don't consider this an issue that would block merging. Just a small improvement for future maintainability. But I have some tech-debt related to this in all the modules. So if you don't have time to clean this up right now, I am ok with adding it to the list for later.

Reply:
Tried to do it, as seemed straightforward enough. Let me know if that's not what you really meant :)